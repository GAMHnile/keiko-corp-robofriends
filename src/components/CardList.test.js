import React from 'react';
import CardList from './CardList';
import { shallow } from 'enzyme';

describe('CardList component tests', ()=>{
    it('renders cardList component', ()=>{
        const mockRobots = [
            {
                id: 1,
                name: "John snow",
                email: "john@SpeechGrammarList.com"
            },
            {
                id: 2,
                name: "jane snow",
                email: "jane@SpeechGrammarList.com"
            },
            {
                id: 3,
                name: "Chris snow",
                email: "chris@SpeechGrammarList.com"
            },
        ]
        const CardListWrapper = shallow(<CardList robots={mockRobots} />);
        expect(CardListWrapper).toMatchSnapshot()
    })
})