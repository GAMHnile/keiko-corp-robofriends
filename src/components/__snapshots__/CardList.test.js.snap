// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component tests renders cardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "john@SpeechGrammarList.com",
          "id": 1,
          "name": "John snow",
        },
        Object {
          "email": "jane@SpeechGrammarList.com",
          "id": 2,
          "name": "jane snow",
        },
        Object {
          "email": "chris@SpeechGrammarList.com",
          "id": 3,
          "name": "Chris snow",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="john@SpeechGrammarList.com"
          id={1}
          name="John snow"
        />,
        <Card
          email="jane@SpeechGrammarList.com"
          id={2}
          name="jane snow"
        />,
        <Card
          email="chris@SpeechGrammarList.com"
          id={3}
          name="Chris snow"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "john@SpeechGrammarList.com",
          "id": 1,
          "name": "John snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "jane@SpeechGrammarList.com",
          "id": 2,
          "name": "jane snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "email": "chris@SpeechGrammarList.com",
          "id": 3,
          "name": "Chris snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="john@SpeechGrammarList.com"
            id={1}
            name="John snow"
          />,
          <Card
            email="jane@SpeechGrammarList.com"
            id={2}
            name="jane snow"
          />,
          <Card
            email="chris@SpeechGrammarList.com"
            id={3}
            name="Chris snow"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "john@SpeechGrammarList.com",
            "id": 1,
            "name": "John snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "jane@SpeechGrammarList.com",
            "id": 2,
            "name": "jane snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "email": "chris@SpeechGrammarList.com",
            "id": 3,
            "name": "Chris snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
